python/lsst/ip/diffim/detectAndMeasure.py

```diff
@@ -376,6 +383,33 @@ def processResults(self, science, matchedTemplate, difference, sources, table,
 
         return measurementResults
 
+    def removeBadSources(self, diaSources):
+        """Remove bad diaSources from the catalog.
+
+        Parameters
+        ----------
+        diaSources : `lsst.afw.table.SourceCatalog`
+            The catalog of detected sources.
+
+        Returns
+        -------
+        diaSources : `lsst.afw.table.SourceCatalog`
+            The updated catalog of detected sources, with any source that has a
+            flag in ``config.badSourceFlags`` set removed.
+        """
+        flags = np.ones(len(diaSources), dtype=bool)
+        for flag in self.config.badSourceFlags:
+            try:
+                flags &= ~diaSources[flag]
+            except Exception as e:
+                self.log.warning("Could not apply source flag: %s", e)
+        nBad = np.count_nonzero(~flags)
+        if nBad > 0:
+            self.log.warning(f"Found and removed {nBad} unphysical sources.")
```

Comment:
No string formatting directly in log messages, use the log `%` formatters: 

https://developer.lsst.io/stack/logging.html#basic-usage-in-python

I'm torn on whether this should be `warning` or `info`: it's doing exactly the thing you asked for which implies `info`, but those things shouldn't have been there in the first place which implies `warning`. I wonder if what we really want is a count of sources with each of the "bad" flags in the loop above as an `info` message, so that we have better granularity about what's going on? Maybe this in the loop?

```
flagged = ~diaSources[flag]
nFlagged = sum(flagged)
if nFlagged > 0:
    self.log.info("Removed %d sources with %s.", nFlagged, flag)
```